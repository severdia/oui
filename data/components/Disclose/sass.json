{"name":"Disclose\n","description":"Similar to an accordion but any number of items can be opened\nclosed independently.","start":{"line":1,"column":1},"end":{"line":18,"column":4},"example":[{"type":"html","description":null,"code":"<div class=\"#{$namespace}disclose is-active\">\n  <a href=\"#\" class=\"#{$namespace}disclose__link link--dark\">\n    <div class=\"#{$namespace}disclose__arrow is-active\">\n      <span class=\"#{$namespace}disclose__symbol\"></span>\n      Title of Disclosure\n    </div>\n  </a>\n  <div class=\"#{$namespace}disclose__content\">\n    <h3>Some Title</h3>\n    <p>Lorem ipsum dolor sit amet, consectetur adipisicing elit. Commodi id voluptas vitae eius expedita alias iste deserunt rndis earum voluptatibus quae, numquam dolorum perspiciatis accusantium corporis, beatae maxime quasi. Tempora.</p>\n  </div>\n</div>"}]}