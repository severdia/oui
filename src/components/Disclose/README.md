# Disclose Component

- Add storybook link
- Add design guidelines