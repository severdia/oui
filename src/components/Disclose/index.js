import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';
import { toClass } from 'recompose';
import { withToggle } from '../../utils/recompose-utils';

const Disclose = ({
  children,
  childrenStyle,
  headerStyle,
  isOpen,
  indented,
  title,
  toggle,
}) => {
  const contentClass = classNames(
    'oui-disclose__content',
    {
      ['border--sides border--bottom']: childrenStyle === 'border',
      ['border--bottom']: childrenStyle === 'divider' && isOpen,
    }
  );
  const linkClass = classNames(
    'oui-disclose__link link--dark soft-half flush',
    {
      ['background--faint display--block']: headerStyle === 'header',
      ['border--all background--faint display--block']: headerStyle === 'header-bordered',
      ['hard--left']: !indented,
    }
  );
  const arrowClass = classNames(
    'oui-disclose',
    {
      ['is-active']: isOpen,
    },
  );
  return (
    <div className={ arrowClass } style={{marginTop: '-1px'}}>
      <a onClick={ toggle } className={ linkClass }>
        <div className='oui-disclose__arrow'>
          <span className="oui-disclose__symbol push-half--right"></span>
          { title }
        </div>
      </a>
      <div className={ contentClass }>
        { isOpen && children }
      </div>
    </div>
  );
};

Disclose.propTypes = {
  children: PropTypes.node.isRequired,
  childrenStyle: PropTypes.string,
  headerStyle: PropTypes.string,
  indented: PropTypes.bool,
  isOpen: PropTypes.bool,
  noBorder: PropTypes.bool,
  title: PropTypes.string.isRequired,
  toggle: PropTypes.func,
};

Disclose.defaultProps = {
  indented: true,
};

export default withToggle(toClass(Disclose));
