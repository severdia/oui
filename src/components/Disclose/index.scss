////
/// Disclose
/// @description Similar to an accordion but any number of items can be opened
/// closed independently.
/// @example[html]
///   <div class="#{$namespace}disclose is-active">
///     <a href="#" class="#{$namespace}disclose__link link--dark">
///       <div class="#{$namespace}disclose__arrow is-active">
///         <span class="#{$namespace}disclose__symbol"></span>
///         Title of Disclosure
///       </div>
///     </a>
///     <div class="#{$namespace}disclose__content">
///       <h3>Some Title</h3>
///       <p>Lorem ipsum dolor sit amet, consectetur adipisicing elit. Commodi id voluptas vitae eius expedita alias iste deserunt rndis earum voluptatibus quae, numquam dolorum perspiciatis accusantium corporis, beatae maxime quasi. Tempora.</p>
///     </div>
///   </div>
////

.#{$namespace}disclose,
.oui-disclose,
%oui-disclose {
  @extend %kill-last-child-margin;

  &__link {
    display: block;
    position: relative;
    margin-bottom: 5px;
  }

  &__symbol {
    width: spacer(1);
    display: inline-block;
    pointer-events: none; // Make clicking the arrow icon still open the disclose object.
    margin-right: spacer(0.5);

    &::before {
      @extend %oui-arrow-inline;
      content: '';
      transform: rotate(-135deg);
    }
  }

  &--right {
    .#{$namespace}disclose__symbol,
    .oui-disclose__symbol {
      float: right;
      transform: rotate(180deg);
      margin-right: 0;
      margin-left: spacer(0.5);
    }
  }

  // By default item is closed.

  &__content {
    display: none;
  }

  &__item {
    margin-bottom: spacer(1);
  }

  &.is-active {

    .#{$namespace}disclose__content,
    .oui-disclose__content {
      display: block;
    }

    .oui-disclose__content > :last-child {
        padding-bottom: 20px;
    }

    .#{$namespace}disclose__symbol,
    .oui-disclose__symbol {
      transform: rotate(90deg);
      top: -4px;
      position: relative;
    }

  }
}
