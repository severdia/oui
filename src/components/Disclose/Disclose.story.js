import React from 'react';

import { storiesOf } from '@storybook/react';
import { withKnobs } from '@storybook/addon-knobs';

import Disclose from './index.js';
import Checkbox from '../Checkbox';

const stories = storiesOf('Disclose', module);
stories
  .addDecorator(withKnobs)
  .addDecorator(story => (
    <div id="root-preview">
      {story()}
    </div>
  ));

stories.add('Default', (() => {
  return (
    <Disclose title='default title'>
      <h3>Some Title</h3>
      <p>Lorem ipsum dolor sit amet, consectetur adipisicing elit. Commodi id voluptas vitae eius
      expedita alias iste deserunt rndis earum voluptatibus quae, numquam dolorum perspiciatis
      accusantium corporis, beatae maxime quasi. Tempora.</p>
    </Disclose>
  );
}));

stories.add('Multiple stacked', (() => {
  return (
    <div>
      <Disclose headerStyle='header-bordered' title='Some Title' childrenStyle='border'>
        <div className='soft--left'>
          <h3>Some Title</h3>
          <p>Lorem ipsum dolor sit amet, consectetur adipisicing elit. Commodi id voluptas vitae eius
          expedita alias iste deserunt rndis earum voluptatibus quae, numquam dolorum perspiciatis
          accusantium corporis, beatae maxime quasi. Tempora.</p>
        </div>
      </Disclose>
      <Disclose headerStyle='header-bordered' title='Some Title'>
        <div className='soft--left'>
          <h3>Some Title</h3>
          <p>Lorem ipsum dolor sit amet, consectetur adipisicing elit. Commodi id voluptas vitae eius
          expedita alias iste deserunt rndis earum voluptatibus quae, numquam dolorum perspiciatis
          accusantium corporis, beatae maxime quasi. Tempora.</p>
        </div>
      </Disclose>
    </div>
  );
}));

stories.add('Header style', (() => {
  return (
    <div>
      <Disclose headerStyle='header' title='Browsers'>
        <div className='soft--left'><a>Safari</a></div>
        <div className='soft--left'><a>Chrome</a></div>
        <div className='soft--left'><a>Opera</a></div>
        <div className='soft--left'><a>IE</a></div>
      </Disclose>
      <Disclose headerStyle='header' title='Some Title' childrenStyle='border'>
        <div className='soft--left'>
          <h3>Some Title</h3>
          <p>Lorem ipsum dolor sit amet, consectetur adipisicing elit. Commodi id voluptas vitae eius
          expedita alias iste deserunt rndis earum voluptatibus quae, numquam dolorum perspiciatis
          accusantium corporis, beatae maxime quasi. Tempora.</p>
        </div>
      </Disclose>
    </div>
  );
}));

stories.add('With divider', (() => {
  return (
    <div>
      <Disclose childrenStyle='divider' title='Some Title'>
        <h3>Some Title</h3>
        <p>Lorem ipsum dolor sit amet, consectetur adipisicing elit. Commodi id voluptas vitae eius
        expedita alias iste deserunt rndis earum voluptatibus quae, numquam dolorum perspiciatis
        accusantium corporis, beatae maxime quasi. Tempora.</p>
      </Disclose>
      <Disclose childrenStyle='divider' title='Some Title'>
        <h3>Some Title</h3>
        <p>Lorem ipsum dolor sit amet, consectetur adipisicing elit. Commodi id voluptas vitae eius
        expedita alias iste deserunt rndis earum voluptatibus quae, numquam dolorum perspiciatis
        accusantium corporis, beatae maxime quasi. Tempora.</p>
      </Disclose>
    </div>
  );
}));

stories.add('With ul', (() => {
  return (
    <div>
      <Disclose childrenStyle='divider' title='Some Title'>
        <ul className="push-half oui-input-list">
          <li>
            <Checkbox
              label="When triggered, deactivate if conditions are not met"
              testSection="configure-view-smart-deactivation_enabled"
            />
          </li>
          <li className="push-double--left">
            <Checkbox
              label="Undo changes when page activates"
              testSection="configure-view-smart-undo_on_deactivation"
            />
            <div className="push-triple--left muted">
              Undoing changes during page activation allows your single page app
              to dynamically apply and remove changes. Learn more about when to
              use this, and why this might be useful.
            </div>
          </li>
        </ul>
      </Disclose>
    </div>
  );
}));
