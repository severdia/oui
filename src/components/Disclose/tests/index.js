import React from 'react';
import { mount, shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';

import Disclose from '../index';

const disclose = shallow(<Disclose title='Some title'> <div>disclose this</div> </Disclose>);


describe('Disclose Component ', () => {
  it('renders correctly', () => {
    expect(shallowToJson(disclose)).toMatchSnapshot();
  });

  it('should add a hard left class to the link when the indented prop is false', () => {
    const component = mount(<Disclose title='Some title' indented={ false }> <div>disclose this</div> </Disclose>);
    expect(component.find('a').hasClass('hard--left')).toBe(true);
  });
});
