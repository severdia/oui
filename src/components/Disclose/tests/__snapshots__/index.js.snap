// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Disclose Component  renders correctly 1`] = `
<withHandlers(Disclose)
  isOpen={false}
  title="Some title"
  toggle={[Function]}
>
   
  <div>
    disclose this
  </div>
   
</withHandlers(Disclose)>
`;
